ctsit/nacculator
# Add module for Neuropath version 11

**What does this change do?**

This change adds processing ability for the Neuropath version 11 form (and changes NACCulator's default `-np` flag to point to version 11 instead of 10). Version 10 processing can still be accessed using the `-np10` flag.

This change also adds the capability for NACCulator to handle NP forms that aren't associated with a visit event (for example, in our Brief Data Set project, which is not longitudinal, there are no visit events specified).

**Why was this change made?**

This change was made to address https://jira-prod.ctsi.ufl.edu/browse/ADRC-3451 which calls for the newest version of the Neuropath form to be NACCulator-processable. We need to be able to handle version 11 in order to submit new data to NACC's system. 


## Verification

_List the steps needed to make sure this thing works. Be precise._

1. To run:
clone nacculator from github
install NACCulator by running: 
$ python3 mvenv .venv
$ source .venv/bin/activate
$ pip3 install wheel
$ pip3 install .
set up nacculator_cfg.ini.example by copying the file and naming it nacculator_cfg.ini
download the Brief Data Set project data from REDCap by doing a full data export and saving that csv file to your computer
run NACCulator on the neuropath data by running:
$ redcap2nacc -np <path/to/your/data.csv >test_np.txt
2. The output should look like a regular NACCulator form output txt file, with the log output to the command terminal
3. If you pass _data with errors_, the output should look like regular NACCulator error reporting in the command terminal
4. Test NACCulator's other available flags on our regular REDCap project data (which has all available data including Neuropath version 10), not the BDS (which hosts only version 11 of the Neuropath form).

- Verify the thing does this: Adds a new option to NACCulator processing without affecting any other existing processes
- Verify the thing does not do that: Passes errors or prints out data in the incorrect columns


## Affirmations

All of these should have a check by them. Any exception requires an explanation.
